### PR TITLE
Fix #13: Add --seed/-s {seed} flag for fixed RNG seeding.

### LINKED ISSUE #13: Add a flag to pass a deterministic seed in combination with `--generate`
Currently `--generate` output is non-reproducible because the random seed is not deterministic.

It would be nice to add a `--seed {seed}`/`-s {seed}` flag to make this deterministic. That way, `.ctd` files can be used to deterministically generate testdata from `generators/generators.yaml` files:

Create (or symlink) `generators/ctd/input_format.ctd`.
Create `generators/ctd/run` shell script with contents `checktestdata --generate --seed $1 ./input_format.ctd`.
Call it from `generators.yaml` using `ctd {seed}`.

### PR DESCRIPTION
**Pull request by RagnarGrootKoerkamp**
Fixes #13.

### CODE REVIEW

**Comment by thijskh** [comment]
Could be added to the testsuite, I think

**Comment by RagnarGrootKoerkamp** [comment]
@thijskh done

File: libchecktestdata.cc [hunk 1]
```diff
@@ -1256,10 +1256,14 @@ void init_checktestdata(std::istream &progstream, int opt_mask)
 	}
 
 	// Initialize random generator
-	struct timespec time;
-	clock_gettime(CLOCK_REALTIME,&time);
-	mpz_class seed = 1000000000 * mpz_class(time.tv_sec) + time.tv_nsec;
-	gmp_rnd.seed(seed);
+	if ( seed == static_cast<unsigned long>(-1)) {
```
Comment (eldering) [inline libchecktestdata.cc:1259]: What about making `seed` a (signed) long? Then this static cast isn't necessary, and also the initial assignment on line 96 is strictly speaking correct, and an uninitialized seed can be distinguished from one specified as `ULONG_MAX`.

**Review by eldering — COMMENTED** [review]
Looks fine to me, with a small suggestion (feel free to argue against it).

File: libchecktestdata.cc (hunk 1 above)
Comment (RagnarGrootKoerkamp) [reply libchecktestdata.cc]: Yes, completely agree.

**Pull request merged**